cmd/ddev/cmd/composer-create.go

```diff
@@ -8,21 +8,20 @@ import (
 	"runtime"
 	"strings"
 
+	"github.com/ddev/ddev/pkg/composer"
+	"github.com/ddev/ddev/pkg/ddevapp"
 	"github.com/ddev/ddev/pkg/fileutil"
 	"github.com/ddev/ddev/pkg/nodeps"
-	"github.com/mattn/go-isatty"
-
-	"github.com/ddev/ddev/pkg/ddevapp"
 	"github.com/ddev/ddev/pkg/output"
 	"github.com/ddev/ddev/pkg/util"
+	"github.com/mattn/go-isatty"
 	"github.com/spf13/cobra"
 )
 
-var composerCreateYesFlag bool
-
 // ComposerCreateCmd handles ddev composer create
 var ComposerCreateCmd = &cobra.Command{
-	Use: "create [args] [flags]",
+	Use:  "create [args] [flags]",
+	Args: cobra.MinimumNArgs(1),
```

Comment:
`cobra.MinimumNArgs(1)` works for the following input

```
ddev composer create laravel/laravel --prefer-dist
```

But it does not work for
```
ddev composer create --prefer-dist laravel/laravel
```

Because cobra reads it as:
```
ddev composer create --prefer-dist=laravel/laravel
```

Perhaps something like the code below can check `osargs` for the real arg:

```go
hasAtLeastOneArg := false
for _, element := range osargs {
	if !strings.HasPrefix(element, "--") && !strings.HasPrefix(element, "-") {
		hasAtLeastOneArg = true
		break
	}
}

if !hasAtLeastOneArg {
	util.Failed("failed to create project: provide a package name to install")
}
```

Reply:
I saw the same problem with exactArgs(1). Thought this would resolve it. Really odd that the flags have to come after the args.

Reply:
Yes, passing flags and arguments to another command complicates things. I looked inside cobra and didn't find anything suitable. That's because when you pass unknown flags, cobra doesn't know if that flag takes any value and treats the next item in the cmd as the value for that flag.